### PR TITLE
Use named Vue component classes

### LINKED ISSUE #1103: Name Vue component classes
Currently we use anonymous classes for Vue components, like this:

```javascript
// components/ui/InfiniteScroll.vue
export default class extends Mixins(ScrollSectionsMixin) { }
```

Naming component classes can help with DX by making them easier to distinguish with Vue.js devtools.

This

```javascript
// components/ui/InfiniteScroll.vue
export default class InfiniteScroll extends Mixins(ScrollSectionsMixin) { }
```

would result in this:

98832706-6b7ea900-243d-11eb-83e1-b24b37dbe621

---

Originally raised in https://github.com/Qiskit/qiskit.org/pull/1099#discussion_r521452233

### PR DESCRIPTION
**Pull request by techtolentino**
Fixes #1103 

Added named components that now show up in Vue dev-tools:

Screen Shot 2020-11-18 at 2 50 30 PM

### CODE REVIEW

File: components/learn/TheLearnHeader.vue [hunk 1]
```diff
@@ -22,5 +22,5 @@ import ThePageHeader from '~/components/ui/ThePageHeader.vue'
 import TypewriterEffect from '~/components/ui/TypewriterEffect.vue'
 
 @Component({ components: { ThePageHeader, TypewriterEffect } })
-export default class extends Vue { }
+export default class ThLearnHeader extends Vue { }
```
Comment (y4izus) [inline components/learn/TheLearnHeader.vue:25]: Typo: ThLearnHeader => Th**e**LearnHeader

File: pages/events/_slug.vue [hunk 2]
```diff
@@ -80,7 +80,7 @@ function getBackgroundUris (background: string): [string, string] {
     }
   }
 })
-export default class extends QiskitPage {
+export default class EventsPage extends QiskitPage {
```
Comment (y4izus) [inline pages/events/_slug.vue:83]: I think `EventPage` or `EventEntry` is more appropriate because this component is for a single event's page

**Review by y4izus — CHANGES_REQUESTED** [review]
Great job Randy! just two small changes :)

**Comment by korgan00** [comment]
I want to suggest something.
The pages could be named as XxxxxPage. On one hand, we have the LandingPage and on the other hand Advocates and Events.
If you ask me what "Advocates" class should do, I think about multiple responsibilities such as a special collection class, a component showing several advocates... But I think AdvocatesPage is clear.
As [Yaiza said](https://github.com/Qiskit/qiskit.org/pull/1130#discussion_r526692115), the _slug pages I like the XxxxEntry option.

**Review by eddybrando — APPROVED** [review]
Nice work! 🙂

**Review by korgan00 — APPROVED** [review]

**Review by korgan00 — APPROVED** [review]
Good job!

**Review by y4izus — APPROVED** [review]

**Pull request merged**